Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -566,6 +566,20 @@ namespace AzToolsFramework
 
                 CreateLink(instanceToCreate->get(), instanceToParentUnder->get().GetTemplateId(), undoBatch.GetUndoBatch(), AZStd::move(patch));
 
+                // Update the parent instance cache so that the new instance doesn't get recreated during propagation
+                if (PrefabDomReference cachedOwningInstanceDom = instanceToParentUnder->get().GetCachedInstanceDom();
```

Comment:
Are the else situations errors that bear being reported to the user ? (Or atleast logged for someone who is debugging)

Reply:
I don't think so. Other places in code that update the cached instance DOM don't require its existence. There's a flag that gets set for propagation to do selective deserialization so it's an option. @srikappa-amzn can you confirm?

Reply:
The cached dom will be null until the first time it goes through deserialization. Here, we are checking upon the cached dom of the parent user wants to instantiate under. Since a valid prefab parent needs to exist and be active in the editor before they can instantiate, it must have already gone through deserialization and the cached dom must be already be non-null. If we want to add a check for that, sure, we can put in an assert here if the cached dom is null.